### PR TITLE
BUGFIX: Capture exceptions when creating new user or updating password

### PR DESCRIPTION
**Pull request by Benjamin-K**
This PR replaces #3901.

**Upgrade instructions**

Creating a user or changing the password will no longer throw an exception with a 500 error page but instead show the error in a flash message. This is done by wrapping `UserService::addUser()` and `UserService::setUserPassword()` with a try-catch block. 

**Review instructions**

This PR changes the exception handling when creating new users or updating passwords by wrapping the used methods (`UserService::addUser()` and `UserService::setUserPassword()`) inside a try catch block and showing an error message (FlashMessage) if there was an exception. 
That way it will be easier to extend Neos and add some password checks without showing a 500 error page in the Neos backend, which is currently the case with [JvMTECH.NeosHardening](https://github.com/jvm-tech/JvMTECH.NeosHardening) (see jvm-tech/JvMTECH.NeosHardening#2) and maybe others.

With the adjusted code you can add checks to the password through an Aspect and simply throw an exception, if the requirements do not pass. 

**Checklist**

- [x] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [x] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [x] Reviewer - The first section explains the change briefly for change-logs
- [x] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Neos/Classes/Controller/Module/Administration/UsersController.php [hunk 1]
```diff
@@ -367,7 +380,19 @@ public function updateAccountAction(Account $account, array $roleIdentifiers, ar
...
+            try {
+                $this->userService->setUserPassword($user, $password);
+            } catch (\Exception $e) {
+                $this->addFlashMessage(
+                    // The shown message must be translated when throwing the actual exception.
+                    $e->getMessage(),
+                    $this->translator->translateById('userSettings.passwordUpdateFailed.title', [], null, null, 'Modules', 'Neos.Neos'),
+                    Message::SEVERITY_ERROR,
+                    [],
+                    1647335912
```
Comment (Sebobo) [inline Neos.Neos/Classes/Controller/Module/Administration/UsersController.php]: The id should be different than the one in the `createAction` L196

File: Neos.Neos/Classes/Controller/Module/User/UserSettingsController.php [hunk 2]
```diff
@@ -143,12 +144,24 @@ public function updateAccountAction(array $password = [])
...
+                    Message::SEVERITY_OK
+                );
+            } catch (\Exception $e) {
+                $this->addFlashMessage(
+                    // The shown message must be translated when throwing the actual exception.
+                    $e->getMessage(),
+                    $this->translator->translateById('userSettings.passwordUpdateFailed.title', [], null, null, 'Modules', 'Neos.Neos'),
+                    Message::SEVERITY_ERROR,
+                    [],
+                    1647335912
```
Comment (Sebobo) [inline Neos.Neos/Classes/Controller/Module/User/UserSettingsController.php:161]: Also use a new id here.

**Review by Sebobo — CHANGES_REQUESTED** [review]
Besides the two tiny issues, looks fine by reading, thx!

File: Neos.Neos/Classes/Controller/Module/Administration/UsersController.php (hunk 1 above)
Comment (Benjamin-K) [reply Neos.Neos/Classes/Controller/Module/Administration/UsersController.php]: Done

**Review by Sebobo — APPROVED** [review]
Looks fine by reading

**Comment by paavo** [comment]
Any plans when this will be merged @Sebobo?

**Review by kdambekalns — APPROVED** [review]
Looks good by reading!

**Review by crydotsnake — APPROVED** [review]

**Pull request merged**